daemon/src/send_future_ext.rs

```diff
@@ -0,0 +1,35 @@
+use async_trait::async_trait;
+use std::error::Error;
+
+#[async_trait]
+pub trait Warn {
+    async fn warn(self);
```

Comment:
Perhaps pass in a `&str` so we  can add a message to the error?

Reply:
```suggestion
    async fn log_failure(self);
```

A bit longer but I think easier to understand.

Reply:
> Perhaps pass in a `&str` so we can add a message to the error?

What does the final log string look like in your mind. I'm not convinced we need this, but maybe it could be an `Option`.

Reply:
I see it as an equivalent to `anyhow::Context`.

Very likely, a fallible handler has more than one `?` in it so the overall context of what bigger operation failed is lost.

What we could do is introduce a trait similar to `ActorName` for messages and include a generic message here that says `"Message handler for {} failed: {:#}"`.

Probably can even unify that to a single `Name` trait or - dare I say it - use `Display` 😁

Reply:
I prefer `Name` here since we probably only want to log the name of the type. If not I would use `Debug`.